dpnp/backend/kernels/dpnp_krnl_linalg.cpp

```diff
@@ -34,47 +34,78 @@
 namespace mkl_blas = oneapi::mkl::blas::row_major;
 namespace mkl_lapack = oneapi::mkl::lapack;
 
-template <typename _DataType>
-class dpnp_cholesky_c_kernel;
 
 template <typename _DataType>
-void dpnp_cholesky_c(void* array1_in, void* result1, size_t* shape)
+void dpnp_cholesky_c(void* array1_in, void* result1, size_t size, size_t last_shape_elem)
```

Comment:
Please, declare `size` and `last_shape_elem` as `const size_t` type.

Reply:
And rename please `last_shape_elem`.
Some variants for you: `step_size`,  `data_size`, `block_size`, `last_dim_size`.